src/Tools/Source/CompilerGeneratorTools/Source/BoundTreeGenerator/Deserializer.cs

```diff
@@ -0,0 +1,220 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Xml.Serialization;
+using System.Xml.Linq;
+using System.Collections;
+using System.IO;
+using System.Threading;
+
+namespace BoundTreeGenerator
+{
+    internal class Deserializer
```

Comment:
It feels like, rather than implementing a custom deserializer and comment handler, it would be much simpler to just turn those comments into XML nodes, like we do for IOperation `<summary>` nodes. It does require updates to the xml doc to convert the existing comments to this new form, but it feels more maintainable long-term.